ow_materials/include/AxisAlignedGrid.h

```diff
@@ -0,0 +1,63 @@
+// The Notices and Disclaimers for Ocean Worlds Autonomy Testbed for Exploration
+// Research and Simulation can be found in README.md in the root directory of
+// this repository.
+
+#ifndef AXIS_ALIGNED_GRID_H
+#define AXIS_ALIGNED_GRID_H
+
+#include <vector>
+#include <memory>
+
+#include <ignition/math/AxisAlignedBox.hh>
+
+namespace ow_materials {
+
+template <typename T>
+class AxisAlignedGrid {
+public:
+  AxisAlignedGrid(double const x0, double const y0, double const z0,
+                  double const x1, double const y1, double const z1,
+                  double const side_length, T const initial_value);
+  ~AxisAlignedGrid() = default;
+
+  AxisAlignedGrid() = delete;
+  AxisAlignedGrid(const AxisAlignedGrid&) = delete;
+  AxisAlignedGrid& operator=(const AxisAlignedGrid&) = delete;
+
+  const ignition::math::Vector3d &getDiagonal() const;
+  const ignition::math::Vector3d &getMaxCorner() const;
+  const ignition::math::Vector3d &getMinCorner() const;
+  const ignition::math::Vector3d &getCenter() const;
+
+  inline T &getCellValue(size_t const i, size_t const j, size_t const k) {
+    return m_cells[index(i, j, k)];
+  };
+
+  inline T const &getCellValue(size_t const i, size_t const j,
+                               size_t const k) const {
+    return m_cells[index(i, j, k)];
+  };
+
+private:
+
+  inline size_t index(size_t const i, size_t const j, size_t const k) const {
+    assert(i < m_dimensions.X());
+    assert(j < m_dimensions.Y());
+    assert(k < m_dimensions.Z());
+    return i + j * m_dimensions.X() + k * m_dimensions.X() * m_dimensions.Y();
```

Comment:
I'm always leery of `assert` in its ability to crash the system.  Is it perhaps overkill for checking dimensions?

Reply:
I get the impression there is a particular time and place for `assert`, but it looks to me our project's build system does not differentiate between debug and release builds, which is where assert is useful. In a release build, an `assert` does nothing because it is only intended as a debug tool. I do not know if our build system builds with `NDEBUG` defined or not. Additionally, there is a chance the public method that provides `i`, `j`, and `k` will be callable by the user, so there is a chance this should be a user input error as opposed to an internal error, which is the only kind `assert` should ever check for. 

I'll remove them for now, but may replace them with an if-statement and exception throw later as I come to better understand who exactly needs to call `getCellValue`.

_Source_ https://stackoverflow.com/questions/12062365/is-using-assert-in-c-bad-practice

Reply:
`assert`s removed

Reply:
Thanks and good question - I don't know if a debug mode as you describe is supported in our build.  You could create a Jira, and this may be a good intern task.

Reply:
I experimented more with asserts in another function, and it turns out they really aren't good to use in Gazebo plugins, and I think that makes sense since they are dynamically loaded libraries. Whenever they are triggered, gazebo will crash, but the message contained by the exception is nowhere to be found in the log file or simulation terminal. Maybe it winds up in another log file, but I think that indicates it's much better to use _only_ Gazebo's logging system when in code operates inside a plugin and to catch absolutely everything that is being thrown.